src/rgw/rgw_rest_s3.cc

```diff
@@ -6492,10 +6493,36 @@ rgw::auth::s3::STSEngine::authenticate(
     return result_t::reject(ret);
   }
   //Authentication
+  std::string secret_access_key;
   //Check if access key is not the same passed in by client
   if (token.access_key_id != _access_key_id) {
```

Comment:
this condition check probably ensures that the system user credentials is used to sign the request _only_ when the zone is secondary? otherwise we can check for` if (!driver->is_meta_master())`

Reply:
the request is forwarded from the secondary to the primary so the check would be to see if the current node is the master which is if (driver->is_meta_master()) ?

Reply:
it looks like you are already checking for `user->get_info().system` to ensure that authenticating against session token credentials is by-passed only for secondary and not for requests originating from primary. that should be enough.

Reply:
ok

Reply:
does user->get_info().system guarantee that the code operates only on the primary? Or to be on the safer side, do you think I must add the check for driver->is_meta_master()